src/Blocks/BlockSponge.h

```diff
@@ -0,0 +1,118 @@
+
+#pragma once
+
+#include "BlockHandler.h"
+
+
+
+
+
+class cBlockSpongeHandler :
+	public cBlockHandler
+{
+public:
+	cBlockSpongeHandler(BLOCKTYPE a_BlockType):
+		cBlockHandler(a_BlockType)
+	{
+	}
+
+
+
+
+
+	virtual void Check(cChunkInterface & a_ChunkInterface, cBlockPluginInterface & a_PluginInterface, int a_RelX, int a_RelY, int a_RelZ, cChunk & a_Chunk) override
+	{
+		if (GetSoaked(Vector3i(a_RelX, a_RelY, a_RelZ), a_Chunk))
+		{
+			return;
+		}
+		cBlockHandler::Check(a_ChunkInterface, a_PluginInterface, a_RelX, a_RelY, a_RelZ, a_Chunk);
+	}
+
+
+
+
+
+	/** Check blocks above and around to see if they are water.
+		If one is, soak this sponge and remove nearby water.
+		Returns TRUE if the block was changed. */
+	bool GetSoaked(Vector3i a_Rel, cChunk & a_Chunk)
+	{
+		static const std::array<Vector3i, 5> WaterCheck
+		{
+			{
+				{ 1, 0,  0},
+				{-1, 0,  0},
+				{ 0, 0,  1},
+				{ 0, 0, -1},
+				{ 0, 1,  0},
+			}
+		};
+		BLOCKTYPE TargetBlock;
+		NIBBLETYPE TargetMeta;
+		a_Chunk.GetBlockTypeMeta(a_Rel.x, a_Rel.y, a_Rel.z, TargetBlock, TargetMeta);
+		if (TargetMeta == E_META_SPONGE_DRY){
+		bool ShouldSoak = std::any_of(WaterCheck.cbegin(), WaterCheck.cend(), [a_Rel, & a_Chunk](Vector3i a_Offset)
+			{
+				BLOCKTYPE NeighborType;
+				return (
```

Comment:
```suggestion
				return IsWet(a_Rel + a_Offset, a_Chunk);
```

Reply:
You can either make `IsWet` static or explicitly add `this` to the lambda capture list.
```cpp
[this, a_Rel, & a_Chunk](Vectir3i Offset) { return IsWet(a_Rel + a_Offset, a_Chunk); };
```